diff --git a/src/ATen/native/nested/NestedTensorTransformerFunctions.cpp b/src/ATen/native/nested/NestedTensorTransformerFunctions.cpp
--- a/src/ATen/native/nested/NestedTensorTransformerFunctions.cpp
+++ b/src/ATen/native/nested/NestedTensorTransformerFunctions.cpp
@@ -27,4 +27,4 @@ Tensor NestedTensor_softmax_dropout_xpu(
       /*mask type */ 1); // NestedTensor_to_mask produces a BxT mask
 }

-} // namespace at::native
\ No newline at end of file
+} // namespace at::native
diff --git a/src/ATen/native/sparse/xpu/SparseBinaryOpIntersection.cpp b/src/ATen/native/sparse/xpu/SparseBinaryOpIntersection.cpp
--- a/src/ATen/native/sparse/xpu/SparseBinaryOpIntersection.cpp
+++ b/src/ATen/native/sparse/xpu/SparseBinaryOpIntersection.cpp
@@ -15,12 +15,12 @@ namespace at::native {

 REGISTER_XPU_DISPATCH(
     mul_sparse_sparse_out_stub,
-    &xpu::mul_sparse_sparse_kernel);
+    &xpu::mul_sparse_sparse_kernel)
 REGISTER_XPU_DISPATCH(
     sparse_mask_intersection_out_stub,
-    &xpu::sparse_mask_intersection_kernel);
+    &xpu::sparse_mask_intersection_kernel)
 REGISTER_XPU_DISPATCH(
     sparse_mask_projection_out_stub,
-    &xpu::sparse_mask_projection_kernel);
+    &xpu::sparse_mask_projection_kernel)

 } // namespace at::native
diff --git a/src/ATen/native/sparse/xpu/SparseTensor.cpp b/src/ATen/native/sparse/xpu/SparseTensor.cpp
--- a/src/ATen/native/sparse/xpu/SparseTensor.cpp
+++ b/src/ATen/native/sparse/xpu/SparseTensor.cpp
@@ -19,6 +19,6 @@ SparseTensor _coalesce_sparse_xpu(const SparseTensor& self) {
   return xpu::coalesce_sparse_kernel(self);
 }

-REGISTER_XPU_DISPATCH(flatten_indices_stub, &xpu::flatten_indices_kernel);
+REGISTER_XPU_DISPATCH(flatten_indices_stub, &xpu::flatten_indices_kernel)

 } // namespace at::native
