diff --git a/cmake/Modules/FindSYCL.cmake b/cmake/Modules/FindSYCL.cmake
--- a/cmake/Modules/FindSYCL.cmake
+++ b/cmake/Modules/FindSYCL.cmake
@@ -209,0 +210,3 @@
+      # Apply the compiler launcher to individual SYCL object compiles.
+      set(SYCL_compiler_launcher ${CMAKE_SYCL_COMPILER_LAUNCHER})
+
@@ -327 +330 @@
-      COMMAND ${CMAKE_SYCL_COMPILER_LAUNCHER} ${SYCL_EXECUTABLE}
+      COMMAND ${SYCL_EXECUTABLE}
