diff --git a/cmake/Modules/FindONEMKL.cmake b/cmake/Modules/FindONEMKL.cmake
--- a/cmake/Modules/FindONEMKL.cmake
+++ b/cmake/Modules/FindONEMKL.cmake
@@ -61,8 +61,3 @@ set(MKL_LIB_NAMES "mkl_sycl_blas" "mkl_sycl_dft" "mkl_sycl_lapack"
-if(WIN32)
-  list(APPEND MKL_LIB_NAMES "mkl_intel_thread")
-  list(TRANSFORM MKL_LIB_NAMES APPEND "_dll.lib")
-else()
   list(APPEND MKL_LIB_NAMES "mkl_gnu_thread")
   list(TRANSFORM MKL_LIB_NAMES PREPEND "lib")
   list(TRANSFORM MKL_LIB_NAMES APPEND ".so")
-endif()
