diff --git a/src/ATen/native/transformers/xpu/flash_attn/sycltla/flash_api.h b/src/ATen/native/transformers/xpu/flash_attn/sycltla/flash_api.h
--- a/src/ATen/native/transformers/xpu/flash_attn/sycltla/flash_api.h
+++ b/src/ATen/native/transformers/xpu/flash_attn/sycltla/flash_api.h
@@ -69,7 +69,7 @@ std::tuple<
     const at::Tensor philox_offset);
 
 // Deprecated: Use mha_fwd instead.
-std::tuple<
+__attribute__((visibility("default"))) std::tuple<
@@ -81,7 +81,8 @@ flash_attention_forward_sycltla(
     const float scale);
 
 // Deprecated: Use mha_bwd instead.
-std::tuple<at::Tensor, at::Tensor, at::Tensor> flash_attention_backward_sycltla(
+__attribute__((visibility("default"))) std::tuple<at::Tensor, at::Tensor, at::Tensor>
+flash_attention_backward_sycltla(
