diff --git a/src/BuildOnLinux.cmake b/src/BuildOnLinux.cmake
index 7ff25da..e5f4f2f 100644
--- a/src/BuildOnLinux.cmake
+++ b/src/BuildOnLinux.cmake
@@ -8,6 +8,9 @@
 
 # Build on Linux
 
+if(USE_SYCLTLA)
+  set(BUILD_SEPARATE_OPS OFF)
+endif()
 
 set(TORCH_XPU_OPS_LIBRARIES)
 set(SYCL_LINK_LIBRARIES_KEYWORD PRIVATE)
@@ -78,7 +81,8 @@ else()
     torch_xpu_ops
     OBJECT
     CXX_SOURCES  ${ATen_XPU_MKL_SRCS} ${ATen_XPU_NATIVE_CPP_SRCS} ${ATen_XPU_XCCL_SRCS}
-    SYCL_SOURCES ${ATen_XPU_SYCL_SRCS} ${ATen_XPU_SYCL_COMM_SRCS})
+    SYCL_SOURCES ${ATen_XPU_SYCL_SRCS} ${ATen_XPU_SYCL_COMM_SRCS}
+                 ${ATen_XPU_SYCLTLA_SRCS})
   if(USE_C10D_XCCL)
     target_compile_definitions(torch_xpu_ops PRIVATE USE_C10D_XCCL)
     target_link_libraries(torch_xpu_ops  PUBLIC torch::xccl)
@@ -92,7 +96,7 @@ else()
   list(APPEND TORCH_XPU_OPS_LIBRARIES torch_xpu_ops)
 endif()
 
-if(USE_SYCLTLA)
+if(USE_SYCLTLA AND BUILD_SEPARATE_OPS)
   set(REPLACE_FLAGS_FOR_SYCLTLA TRUE)
   set_build_flags()
 
