diff --git a/src/BuildOnLinux.cmake b/src/BuildOnLinux.cmake
index 042fdb6..ae3da83 100644
--- a/src/BuildOnLinux.cmake
+++ b/src/BuildOnLinux.cmake
@@ -8,6 +8,10 @@
 
 # Build on Linux
 
+if(USE_SYCLTLA)
+  set(BUILD_SEPARATE_OPS OFF)
+endif()
+
 set(TORCH_XPU_OPS_LIBRARIES)
 set(SYCL_LINK_LIBRARIES_KEYWORD PRIVATE)
 
@@ -77,7 +81,13 @@ else()
     torch_xpu_ops
     OBJECT
     CXX_SOURCES  ${ATen_XPU_MKL_SRCS} ${ATen_XPU_NATIVE_CPP_SRCS} ${ATen_XPU_XCCL_SRCS}
-    SYCL_SOURCES ${ATen_XPU_SYCL_SRCS} ${ATen_XPU_SYCL_COMM_SRCS})
+    SYCL_SOURCES ${ATen_XPU_SYCL_SRCS} ${ATen_XPU_SYCL_COMM_SRCS}
+                 ${ATen_XPU_SYCLTLA_SRCS})
+  if(USE_SYCLTLA)
+    target_compile_definitions(torch_xpu_ops PRIVATE ${SYCLTLA_COMPILE_DEFINITIONS})
+    target_include_directories(torch_xpu_ops PRIVATE ${SYCLTLA_INCLUDE_DIRS})
+    target_include_directories(torch_xpu_ops PRIVATE ${ATen_XPU_FLASH_ATTN_DIRS})
+  endif()
   if(USE_C10D_XCCL)
     target_compile_definitions(torch_xpu_ops PRIVATE USE_C10D_XCCL)
     target_link_libraries(torch_xpu_ops  PUBLIC torch::xccl)
@@ -91,7 +101,7 @@ else()
   list(APPEND TORCH_XPU_OPS_LIBRARIES torch_xpu_ops)
 endif()
 
-if(USE_SYCLTLA)
+if(USE_SYCLTLA AND BUILD_SEPARATE_OPS)
   set(REPLACE_FLAGS_FOR_SYCLTLA TRUE)
   set_build_flags()
 
